enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -130,6 +130,8 @@ var (
 			return 11
 		end
 
+		redis.call("hincrby", KEYS[1], "attemptCount", 1)
```

Comment:
sanity check, will this do the right thing if this particular redis.call fails, or if this succeeds but the redis.call right afterwards fails?

Reply:
Practically speaking, these can only fail due to an internal Redis error. If that were to occur, redis.call would terminate the script and return the error to the app. The worst case that can happen is that we incorrectly increment the attempt count by one if the redis.call below fails, but the odds of that actually happening are virtually zero.